### PR TITLE
functions: skip triggers when restoring

### PR DESCRIPTION
**Pull request by lsenta**

### CODE REVIEW

File: functions/src/backup.ts [hunk 1]
```diff
@@ -174,6 +189,41 @@ function reEncodeObject(x: any): any {
...
 
+/**
+ * Set the restore timestamp to now.
+ */
+const setRestoreFlag = async () => {
+  return db.collection('_restore').doc('_DOC').set({ restoredAt: admin.firestore.FieldValue.serverTimestamp() });
+};
+
+const isRestoring = async () => {
+  const x = await db.collection('_restore').doc('_DOC').get();
```
Comment (GrandSchtroumpf) [inline functions/src/backup.ts]: Pas sûr de comprendre ce qu'est "x" en lisant le code. Peut-être trouver un meilleur nom de variable.

File: functions/src/backup.ts [hunk 2]
```diff
@@ -174,6 +189,41 @@ function reEncodeObject(x: any): any {
...
+
+    const now = admin.firestore.Timestamp.now();
+
+    // If we started the restoration less than twenty seconds ago, we are still in the restore process,
+    return restoredAt.toMillis() + THIRTY_SECONDS_IN_MS > now.toMillis();
+  }
+  return false;
+};
+
+export const skipWhenRestoring = (f: any) => {
```
Comment (GrandSchtroumpf) [inline functions/src/backup.ts]: Maybe take the time to type it when you have some.

File: functions/src/backup.ts (hunk 1 above)
Comment (lsenta) [reply functions/src/backup.ts]: FIXED

File: functions/src/backup.ts (hunk 2 above)
Comment (lsenta) [reply functions/src/backup.ts]: Added as a TODO! (I'll look into replacing this solution by something more reliable after Cannes)

**Pull request merged**